# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/lvm2/lv-def-name.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Use more common and shorter lv%d names instead of the clunky lvol%d.

--- LVM2.2.03.15/lib/metadata/lv_manip.c.vanilla	2023-01-19 16:04:35.750961688 +0100
+++ LVM2.2.03.15/lib/metadata/lv_manip.c	2023-01-19 16:05:04.232962649 +0100
@@ -4878,9 +4878,9 @@
 {
 	char uniq_name[NAME_LEN];
 
-	/* If the name is in use, generate new lvol%d */
+	/* If the name is in use, generate new lv%d */
 	if (lv_name_is_used_in_vg(lv->vg, new_name, NULL)) {
-		if (!generate_lv_name(lv->vg, "lvol%d", uniq_name, sizeof(uniq_name))) {
+		if (!generate_lv_name(lv->vg, "lv%d", uniq_name, sizeof(uniq_name))) {
 			log_error("Failed to generate unique name for unused logical volume.");
 			return 0;
 		}
@@ -8453,7 +8453,7 @@
 			return_NULL;
 	}
 
-	if (!(lv = lv_create_empty(new_lv_name ? : "lvol%d", NULL,
+	if (!(lv = lv_create_empty(new_lv_name ? : "lv%d", NULL,
 				   status, lp->alloc, vg)))
 		return_NULL;
 
--- LVM2.2.03.15/lib/metadata/pool_manip.c.vanilla	2023-01-19 16:04:35.734961688 +0100
+++ LVM2.2.03.15/lib/metadata/pool_manip.c	2023-01-19 16:04:55.735962362 +0100
@@ -861,9 +861,9 @@
 	}
 	*c = 0;
 
-	/* If the name is in use, generate new lvol%d */
+	/* If the name is in use, generate new lv%d */
 	if (lv_name_is_used_in_vg(vg, new_name, NULL) &&
-	    !generate_lv_name(vg, "lvol%d", new_name, sizeof(new_name))) {
+	    !generate_lv_name(vg, "lv%d", new_name, sizeof(new_name))) {
 		log_error("Failed to generate unique name for "
 			  "pool metadata spare logical volume.");
 		return 0;
--- LVM2.2.03.15/lib/metadata/writecache_manip.c.vanilla	2023-01-19 16:04:35.720961687 +0100
+++ LVM2.2.03.15/lib/metadata/writecache_manip.c	2023-01-19 16:04:51.227962210 +0100
@@ -142,7 +142,7 @@
 	/*
 	 * Rename lv_fast back to its original name, without the _cvol
 	 * suffix that was added when lv_fast was attached for caching.
-	 * If the name is in use, generate new lvol%d.
+	 * If the name is in use, generate new lv%d.
 	 * Failing to rename is not really a problem, so we intentionally
 	 * do not consider some things here as errors.
 	 */
@@ -155,7 +155,7 @@
 
 	*suffix = 0;
 	if (lv_name_is_used_in_vg(vg, cvol_name, NULL) &&
-	    !generate_lv_name(vg, "lvol%d", cvol_name, sizeof(cvol_name))) {
+	    !generate_lv_name(vg, "lv%d", cvol_name, sizeof(cvol_name))) {
 		log_warn("Failed to generate new unique name for unused LV %s", lv_fast->name);
 		return;
 	}
